pkg/cmd/server/start/start_allinone.go

```diff
@@ -205,38 +207,21 @@ func (o AllInOneOptions) Complete() error {
 // 4.  If only writing configs, it exits
 // 5.  Waits forever
 func (o AllInOneOptions) StartAllInOne() error {
-	if !o.WriteConfigOnly {
+	if !o.ConfigDir.Provided() {
```

Comment:
assign to a var so the intent is clear: `writeConfigOnly := o.ConfigDir.Provided()`